fedora/clients/fasjson.py

```diff
@@ -101,3 +101,51 @@ async def get_users_by_matrix_id(self, matrix_id: str) -> dict | str:
             )
 
         return searchresult[0]
+
+    async def get_user_groups(self, username, params=None):
+        """
+        Retrieves all groups a user belongs to, including membership type (member or sponsor).
+
+        Args:
+            username (str): Username of the user to query.
+            params (dict, optional): Additional query parameters to include in the request.
+
+        Returns:
+            List[dict] or None:
+                - A list of dictionaries containing group information:
+                    - groupname (str): Name of the group.
+                    - membership_type (str): "member" or "sponsor" depending on the user's role.
+                - None (if the user is not found or there's an error).
```

Comment:
If a user isn't found or if there's some other error the function raises an exception and doesn't return anything at all (which I think is good!) so you should drop the `None` case.

Reply:
> If a user isn't found or if there's some other error the function raises an exception and doesn't return anything at all (which I think is good!) so you should drop the `None` case.

Done! Thank you